src/util/string.h

```diff
@@ -26,4 +31,58 @@ class StringCollator {
     QCollator m_collator;
 };
 
-#endif // MIXXX_STRING_H
+/// A nullptr-safe variant of the corresponding standard C function.
+///
+/// Treats nullptr like an empty string and returns 0.
+inline std::size_t strnlen(
+        const char* str,
+        std::size_t len) {
+    if (str == nullptr) {
+        return 0;
+    }
+    // Invoke the global function
+    return ::strnlen(str, len);
+}
+
+/// A nullptr-safe variant of the corresponding standard C function.
+///
+/// Treats nullptr like an empty string and returns 0.
+inline std::size_t wcsnlen(
+        const wchar_t* wcs,
+        std::size_t len) {
+    if (wcs == nullptr) {
+        return 0;
+    }
+    // Invoke the global function
+    return ::wcsnlen(wcs, len);
+}
+
+/// Convert a wide-character C string to QString.
+///
+/// We cannot use Qts wchar_t functions, since they may work or not
```

Comment:
```suggestion
/// We cannot use Qts wchar_t functions, since it may work or not
```

Reply:
Using `it` to refer to multiple functions looks wrong, but you're the native speaker here.

Reply:
Oh, you're right. Nevermind.